### PR TITLE
React 2 week 1 Homework

### PR DESCRIPTION
**Pull request by Nusrath-Jahan**

### CODE REVIEW

File: React-2/Week-1/my-app/app/pages/blog_website/[id].js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    ? id.replace(/-/g, " ").replace(/\b\w/g, (char) => char.toUpperCase())
+    : "";
+
+  return (
+    <div>
+      <h1>{title}</h1>
+      <p>This is the blog post: {title}</p>
+    </div>
+  );
+}
```
Comment (markitosha) [inline React-2/Week-1/my-app/app/pages/blog_website/[id].js:19]: Code of the component is (almost) fine, but the page doesn't work in the end.

image

Why?

Any file in `/app` folder (except _special_ ones like `page.js` or `layout.js`) are considered just components, and will be never rendered if not included to some page, and will not affect path in any way.

The only way to create a path is to create a **folder**. And inside the folder -- a `page.js`. So it suppose to be folder `[id]` and this file suppose to be `page.js` and be inside folder `[id]`

File: React-2/Week-1/my-app/app/pages/blog_website/[id].js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// pages/blogs/[id].js
+import { useRouter } from "next/router";
+
+export default function BlogPost() {
+  const router = useRouter();
+  const { id } = router.query;
```
Comment (markitosha) [inline React-2/Week-1/my-app/app/pages/blog_website/[id].js:6]: 1. To get query from url we use `useSearchParams`
2. To get params from path (as it suppose to be here) we use either `await params` for Server Components or `useParams` for Client Components
3. Router doesn't have `.query`, unfortunatelly

**Review by markitosha — COMMENTED** [review]
Great job!
Left a comment about `blog` task (why it didn't work and how to fix it). Otherwise -- very good

File: React-2/Week-1/my-app/app/pages/blog_website/[id].js (hunk 2 above)
Comment (Nusrath-Jahan) [reply React-2/Week-1/my-app/app/pages/blog_website/[id].js:6]: Would you kindly review my code once more?

File: React-2/Week-1/my-app/app/pages/blog_website/[id].js (hunk 1 above)
Comment (Nusrath-Jahan) [reply React-2/Week-1/my-app/app/pages/blog_website/[id].js:19]: image

**Pull request merged**